x/gov/keeper/msg_server.go

```diff
@@ -31,6 +35,13 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *types.MsgSubmitPro
 		return nil, err
 	}
 
+	bytes, err := proposal.Marshal()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx.BlockGasMeter().ConsumeGas(uint64(3*gasCostPerStorage*len(bytes)), "Submit proposal")
```

Comment:
any thought on this gas value?
cc: @robert-zaremba

Reply:
How did we come up with this value? Personally, i think a min self-deposit approach is better because we can then just tweak the param.

Reply:
We are automatically deleting it - so the motivation here is to pay for that operation. Deposit doesn't really pay for the operation.

Reply:
@atheeshp  could you link the issue next to the line to explain why we do it?

Reply:
Yes, the initial deposit does @robert-zaremba

Reply:
The deposit is returned. Here we want to consume a gas for the `EndBlocker` operations.

Reply:
I still don't think this is a clean approach, but I can't think of a more streamlined way atm.

Reply:
> Here we want to consume a gas for the EndBlocker operations.

Shouldn't we then charge exactly what EndBlocker consumes? if I'm not mistaken, it would be `gas_cost_per_READ * len(proposal_bytes * {the number of blocks where it gets read}`

Reply:
I think this is what we are charging for: submit proposal, inactive queue operation, active queue operation.